edge-util/src/Microsoft.Azure.Devices.Edge.Util/BackgroundTask.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public static class BackgroundTask
+    {
+        static readonly ConcurrentDictionary<string, BackgroundTaskStatus> TaskStatuses = new ConcurrentDictionary<string, BackgroundTaskStatus>();
+
+        public static (string correlationId, BackgroundTaskStatus backgroundTaskStatus) Run(Func<Task> task, string operation, CancellationToken cancellationToken)
+        {
+            var correlationId = Guid.NewGuid().ToString();
+            BackgroundTaskStatus backgroundTaskStatus = new BackgroundTaskStatus(BackgroundTaskRunStatus.Running, operation);
+            TaskStatuses.TryAdd(correlationId, backgroundTaskStatus);
```

Comment:
You aren't checking the return value of `TryAdd` (and `TryUpdate` below); does it matter if it returned false?

Reply:
yeah, good point, I forgot to change that. Let me do that. One sec.